src/axom/core/utilities/FileUtilities.hpp

```diff
@@ -119,6 +121,97 @@ void getDirName(std::string& dir, const std::string& path);
  */
 int removeFile(const std::string& filename);
 
+/**
+ * /brief Utility class for managing temporary files.
+ *
+ * The TempFile class provides a convenient way to create and manage temporary files.
+ * It ensures that the file is deleted upon destruction, if specified.
+ * The class is non-copyable to prevent accidental duplication of file handles.
+ *
+ * \note The path to the temp file is likely different from the supplied \a file_name
+ */
+class TempFile
+{
+public:
+  /**
+   * \brief Creates a tmp file using \a file_name and \a file_ext
```

Comment:
Minor nit: Some comments use `tmp`, some use `temp`, and others use `temporary`.  I find this distracting. Please ignore if others don't.

Reply:
Thanks -- I'll update it

Reply:
Done